File: konserkita_mobile/lib/models/ticket.dart
import 'ticket_type.dart';

class Ticket {
  final int id;
  final int transactionId;
  final int ticketTypeId;
  final int userId;
  final String ticketCode;
  final bool isUsed;
  final String? usedAt;
  final String? seatCode;
  final TicketType? ticketType;

  Ticket({
    required this.id,
    required this.transactionId,
    required this.ticketTypeId,
    required this.userId,
    required this.ticketCode,
    required this.isUsed,
    this.usedAt,
    this.seatCode,
    this.ticketType,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) {
    return Ticket(
      id: json['id'],
      transactionId: json['transaction_id'],
      ticketTypeId: json['ticket_type_id'],
      userId: json['user_id'],
      ticketCode: json['ticket_code'],
      isUsed: json['is_used'] == 1 || json['is_used'] == true,
      usedAt: json['used_at'],
      seatCode: json['seat'] != null ? json['seat']['seat_code'] : null,
      ticketType: json['ticket_type'] != null ? TicketType.fromJson(json['ticket_type']) : null,
    );
  }
}

File: konserkita_api/app/Models/Ticket.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Ticket extends Model
{
    use HasFactory;

    protected $fillable = [
        'transaction_id',
        'ticket_type_id',
        'user_id',
        'ticket_code',
        'is_used',
        'used_at',
        'is_cancelled',
        'seat_id',
    ];

    protected $casts = [
        'is_used' => 'boolean',
        'is_cancelled' => 'boolean',
        'used_at' => 'datetime',
    ];

    public function transaction(): BelongsTo
    {
        return $this->belongsTo(Transaction::class);
    }

    public function ticketType(): BelongsTo
    {
        return $this->belongsTo(TicketType::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function seat(): BelongsTo
    {
        return $this->belongsTo(Seat::class);
    }
}
